### PR TITLE
WIP first draft of joint RHEL7/ RHEL8 image

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
Needs to be built on a RHEL8 machine with access to RHEL8 repositories ( I built on a beaker machine)

build using:
podman build -t quay.io/jtanenba/sdn:latest -f images/sdn/Dockerfile.mine .

pushed to a repo using
podman push quay.io/jtanenba/sdn

--change to your own repos --

and deployed it in a running 4.0 cluster with info from the following guide
https://github.com/openshift/cluster-network-operator/blob/master/INSTALLER-HACKING.md#building-a-development-sdn-image

### CODE REVIEW

**Comment by smarterclayton** [comment]
How big is the final image?

**Comment by JacobTanenbaum** [comment]
1.03 GB

File: images/sdn/Dockerfile.mine [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    vendor/github.com/containernetworking/plugins/plugins/ipam/host-local \
+    vendor/github.com/containernetworking/plugins/plugins/main/loopback \
+    "
+
+FROM registry.svc.ci.openshift.org/ocp/4.0:base AS RHEL7
+COPY mirror/ocp-4.0-default.repo /etc/yum.repos.d/ocp-4.0-default.repo
+COPY mirror/ops-mirror.pem /etc/yum.repos.d/ops-mirror.pem
+COPY --from=builder /go/src/github.com/openshift/origin/_output/local/bin/linux/amd64/openshift-sdn /usr/bin/
+COPY --from=builder /go/src/github.com/openshift/origin/_output/local/bin/linux/amd64/sdn-cni-plugin /opt/cni/bin/openshift-sdn
+COPY --from=builder /go/src/github.com/openshift/origin/_output/local/bin/linux/amd64/loopback /opt/cni/bin/
```
Comment (pecameron) [inline images/sdn/Dockerfile.mine]: does teh build create loopback and host-local? Its not in your make above.

File: images/sdn/Dockerfile.mine [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+COPY --from=builder /go/src/github.com/openshift/origin/_output/local/bin/linux/amd64/host-local /opt/cni/bin/
+RUN INSTALL_PKGS=" \
+      openvswitch2.11 container-selinux socat ethtool iptables nmap-ncat \
+      libmnl libnetfilter_conntrack conntrack-tools \
+      libnfnetlink iproute bridge-utils procps-ng openssl \
+      binutils xz sysvinit-tools dbus \
+      " && \
+    rm /etc/yum.repos.d/content.repo && \
+    yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
+    mkdir -p /etc/sysconfig/cni/net.d && \
```
Comment (pecameron) [inline images/sdn/Dockerfile.mine]: do you know why this is needed? I don't remember /etc/sysconfig/cni/net.d in the daemonset mounts. I have been using /etc/cni/net.d (I am trying to figure out how all this works)

**Pull request closed without merging**